tests/framework/results/test_context.py

```diff
@@ -196,14 +206,224 @@ def test_add_observation_nop_stratifications(default, additional, excluded, matc
     default_stratifications,
     additional_stratifications,
     excluded_stratifications,
-    expected_groupers,
+    expected_stratifications,
 ):
     ctx = ResultsContext()
     # default_stratifications would normally be set via ResultsInterface.set_default_stratifications()
     ctx._default_stratifications = default_stratifications
-    groupers = ctx._get_groupers(additional_stratifications, excluded_stratifications)
-    assert sorted(groupers) == sorted(expected_groupers)
+    stratifications = ctx._get_stratifications(
+        additional_stratifications, excluded_stratifications
+    )
+    assert sorted(stratifications) == sorted(expected_stratifications)
+
+
+@pytest.mark.parametrize(
+    "name, pop_filter, aggregator_sources, aggregator, stratifications",
+    [
+        ("wizard_count", "tracked==True", None, len, ["house", "familiar"]),
+        ("power_level_total", "tracked==True", ["power_level"], sum, ["house", "familiar"]),
+        (
+            "wizard_time",
+            "tracked==True",
+            [],
+            _aggregate_state_person_time_function,
+            ["house", "familiar"],
+        ),
+        ("wizard_count", "tracked==True", None, len, ["house"]),
+        ("power_level_total", "tracked==True", ["power_level"], sum, ["house"]),
+        (
+            "wizard_time",
+            "tracked==True",
+            [],
+            _aggregate_state_person_time_function,
+            ["house"],
+        ),
+    ],
+    ids=[
+        "len_aggregator_two_stratifications",
+        "sum_aggregator_two_stratifications",
+        "custom_aggregator_two_stratifications",
+        "len_aggregator_one_stratification",
+        "sum_aggregator_one_stratification",
+        "custom_aggregator_one_stratification",
+    ],
+)
+def test_gather_results(name, pop_filter, aggregator_sources, aggregator, stratifications):
+    """Test cases where every stratification is in gather_results"""
+    ctx = ResultsContext()
+
+    # Generate population DataFrame
+    population = BASE_POPULATION.copy()
+    # Mock out some extra columns that would be produced by the manager's _prepare_population() method
+    population["current_time"] = pd.Timestamp(year=2045, month=1, day=1, hour=12)
+    population["step_size"] = timedelta(days=28)
+    population["event_time"] = pd.Timestamp(year=2045, month=1, day=1, hour=12) + timedelta(
+        days=28
+    )
+    event_name = "collect_metrics"
+
+    # Set up stratifications
+    if "house" in stratifications:
+        ctx.add_stratification("house", ["house"], CATEGORIES, None, True)
+    if "familiar" in stratifications:
+        ctx.add_stratification("familiar", ["familiar"], FAMILIARS, None, True)
+    ctx.add_observation(
+        name,
+        pop_filter,
+        aggregator_sources,
+        aggregator,
+        stratifications,
+        [],
+        event_name,
+    )
+
+    i = 0
+    for _ in ctx.gather_results(population, event_name):
+        i += 1
+    assert i == 1
```

Comment:
I don't understand what this assert is demonstrating. Can you add some comments to explain why this tests that `gather_results` works?

Shouldn't we also test that the observed values are correct?

Reply:
It's demonstrating that we get a return value. It's not checking for correctness of the applied `aggregator` but that could be added.